cypress/e2e/genreSelect.cy.ts

```diff
@@ -0,0 +1,18 @@
+describe("GenreSelect", () => {
+  it("selects a genre and triggers callback with correct genre", () => {
+    const selectedGenre = "HORROR";
+    cy.visit("http://localhost:3000/", {
+      onBeforeLoad(win) {
+        cy.stub(win.console, "log").as("consoleLog");
+        cy.stub(win.console, "error").as("consoleError");
+      },
+    });
+
+    cy.contains("button", selectedGenre).click();
+
+    //this will not work with css module
+    //cy.contains('button', selectedGenre).should('have.class', 'selected');
+
+    cy.get("@consoleLog").should("be.calledWith", selectedGenre);
```

Comment:
better to use spy than console log:
 const onSelect = cy.spy();

Reply:
Here I have problem how to use Spy. How can I check if function was called. I found some cv.Mount but wasn't able to make it run.

Reply:
Ok. There are some workarounds how to do it.
But. This is the area of unit-tests (check if particular function wass called)

In this case we need to undestand if the coresponding button is selected after the click:
1) we can rely on class:

`    cy.get('[data-cy="comedy"]').click();
    cy.get('[data-cy="comedy"]')
  .invoke('attr', 'class')
  .should('match', /GenreSelect_selected__/);`

2) we can rely on css property (not selected doesnt have the border bottom):
`cy.get('[data-cy="comedy"]').should('have.css', 'border-bottom');`

As you see it is usefull to add data attribute to the button in the component( smth like this: 
`        <button
          key={genre}
          onClick={() => onSelect(genre)}
          data-cy={genre.toLowerCase()}
          className=.....
        >
          {genre}
        </button>`
)